tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/sequencefinder/OnlineShopDFDTest.java

```diff
@@ -11,86 +12,113 @@
 import org.dataflowanalysis.analysis.dfd.core.DFDFlowGraphCollection;
+import org.dataflowanalysis.analysis.dfd.core.DFDTransposeFlowGraphFinder;
 import org.dataflowanalysis.analysis.dfd.core.DFDVertex;
 import org.dataflowanalysis.examplemodels.Activator;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 public class OnlineShopDFDTest {
 
-    private DFDConfidentialityAnalysis analysis;
+    private DFDConfidentialityAnalysis cylcicAnalysis;
+    private DFDConfidentialityAnalysis linearAnalysis;
 
     @BeforeEach
     public void initAnalysis() {
-        final var dataFlowDiagramPath = Paths.get("models", "OnlineShopDFD", "onlineshop.dataflowdiagram");
-        final var dataDictionaryPath = Paths.get("models", "OnlineShopDFD", "onlineshop.datadictionary");
+        final var dataFlowDiagramPath = Paths.get("models", "OnlineShopDFD", "onlineshop.dataflowdiagram")
+                .toString();
+        final var dataDictionaryPath = Paths.get("models", "OnlineShopDFD", "onlineshop.datadictionary")
+                .toString();
 
-        this.analysis = new DFDDataFlowAnalysisBuilder().standalone()
+        cylcicAnalysis = new DFDDataFlowAnalysisBuilder().standalone()
                 .modelProjectName(TEST_MODEL_PROJECT_NAME)
                 .usePluginActivator(Activator.class)
-                .useDataFlowDiagram(dataFlowDiagramPath.toString())
-                .useDataDictionary(dataDictionaryPath.toString())
+                .useDataFlowDiagram(dataFlowDiagramPath)
+                .useDataDictionary(dataDictionaryPath)
+                .useTransposeFlowGraphFinder(DFDCyclicTransposeFlowGraphFinder.class)
                 .build();
 
-        this.analysis.initializeAnalysis();
+        linearAnalysis = new DFDDataFlowAnalysisBuilder().standalone()
+                .modelProjectName(TEST_MODEL_PROJECT_NAME)
+                .usePluginActivator(Activator.class)
+                .useDataFlowDiagram(dataFlowDiagramPath)
+                .useDataDictionary(dataDictionaryPath)
+                .useTransposeFlowGraphFinder(DFDTransposeFlowGraphFinder.class)
+                .build();
+
+        cylcicAnalysis.initializeAnalysis();
+        linearAnalysis.initializeAnalysis();
     }
 
     @Test
-    public void numberOfTransposeFlowGraphs_equalsThree() {
+    public void testCyclicAnalysis() {
+        runAllChecksForAnalysis(cylcicAnalysis);
+    }
+
+    @Test
+    public void testLinearAnalysis() {
+        runAllChecksForAnalysis(linearAnalysis);
+    }
+
+    private void runAllChecksForAnalysis(DFDConfidentialityAnalysis analysis) {
+        checkNumberOfTransposeFlowGraphs(analysis, 3);
+        checkSinks(analysis, List.of("User", "Database", "Database"));
+        checkNodeLabels(analysis, "User", List.of("EU"));
+        checkDataLabelPropagation(analysis, "User", List.of("Public"));
+        checkRealisticConstraints(analysis, 1, 0);
+        checkIsNotCyclic(analysis);
+    }
```

Comment:
This causes finding of transpose flow graphs multiple times, the flow graph collection should be passed as parameter instead

Reply:
Other comment, why not extract all the checks into separate test cases, so we may get more information when the test cases fail

Reply:
Regarding the first, the old dfd test also did this.

Rgarding the second: We have two analyses on which to run the test, this way has the least amount of code publication i think

Reply:
Regarding 1.: While that may be the case, the performance impact here is way more noticeable as a multitude of analyses are run

Reply:
Regarding 2.: While we gain a tiny amount of code duplication (e.g. the setup) we gain a better understanding of what is wrong when a test case fails

Reply:
Point 1 is adressing, still not found a good way for point 2